db/python/tables/sequence.py

```diff
@@ -279,21 +279,45 @@ async def get_sequence_type_numbers_for_project(self, project: ProjectId):
         """
 
         _query = """
-SELECT type, COUNT(*) as n
-FROM (
-    SELECT sq.type
-    FROM sample_sequencing sq
-    INNER JOIN sample s ON s.id = sq.sample_id
-    WHERE s.project = :project
-    GROUP BY s.id, sq.type
-) as s
-GROUP BY type
+            SELECT type, COUNT(*) as n
+            FROM (
+                SELECT sq.type
+                FROM sample_sequencing sq
+                INNER JOIN sample s ON s.id = sq.sample_id
+                WHERE s.project = :project
+                GROUP BY s.id, sq.type
+            ) as s
+            GROUP BY type
         """
 
         rows = await self.connection.fetch_all(_query, {'project': project})
 
         return {r['type']: r['n'] for r in rows}
 
+    async def get_sequence_type_numbers_by_batch_for_project(self, project: ProjectId):
+        """
+        This groups by batch. Python NONE types and SQL NULL types are combined into a null string by IFNULL
+        """
+
+        _query = """
+            SELECT batch, type, count(*) AS n
+            FROM (
+                SELECT IFNULL(JSON_EXTRACT(sq.meta, '$.batch'), 'null') as batch, sq.type
+                FROM sample_sequencing sq
+                INNER JOIN sample s ON s.id = sq.sample_id
+                WHERE s.project = :project
+            ) as p
+            GROUP BY batch, type
```

Comment:
You can actually use the selected fields inside the group by to avoid the double query, eg:

```suggestion
            SELECT IFNULL(JSON_EXTRACT(sq.meta, '$.batch'), 'null') as batch, sq.type, COUNT(*) AS n
            FROM sample_sequencing sq
            INNER JOIN sample s ON s.id = sq.sample_id
            WHERE s.project = :project
```